### PR TITLE
Prepare very first release as ownCloud app

### PR DESCRIPTION
**Pull request by DeepDiver1975**

### CODE REVIEW

File: src/Phoenix.vue [hunk 1]
```diff
@@ -24,7 +24,9 @@ export default {
 		'login'     : Login
 	},
 	beforeMount () {
-		this.$client.setInstance(this.$root.config.server);
+		let instance = this.$root.config.server || window.location.origin;
+		console.error(instance);
```
Comment (PVince81) [inline src/Phoenix.vue]: leftovers?

File: Makefile.release [hunk 2]
```diff
@@ -0,0 +1,68 @@
+SHELL := /bin/bash
```
Comment (PVince81) [inline Makefile.release]: adding a new rule to the existing makefile might be better.

this is ok as a quick solution, we can make it nicer later

Comment (DeepDiver1975) [reply Makefile.release]: I'm following the idea of @patrickjahns where two makefiles exist: one for dev and one for release. To be discussed ....

**Comment by DeepDiver1975** [comment]
> try is in the default apps folder ....

there is another issue involved - I'll work on this on Monday ...

**Comment by michaelstingl** [comment]
Argh – I‘ll try to work around this… (Docker setup always uses non-default apps directory)

**Comment by DeepDiver1975** [comment]
> Argh – I‘ll try to work around this… (Docker setup always uses non-default apps directory)

But there are more issues - related to the server setup ... I need to debug this


Maybe even a slim php code snippet is needed to make this reliably work as part of an owncloud app ..... I need to dig deeper ...

**Comment by michaelstingl** [comment]
Does it also rely on the index.php in the URL? (default Docker setup has the nicer URL's without index.php)

I click the Phoenix icon: `GET http://localhost/apps/phoenix/`

It sends me back:

```
HTTP/1.1 302 Found
Date: Sat, 15 Sep 2018 08:50:09 GMT
Server: Apache
Strict-Transport-Security: max-age=15768000; preload
Expires: Thu, 19 Nov 1981 08:52:00 GMT
Cache-Control: no-store, no-cache, must-revalidate
Pragma: no-cache
Content-Security-Policy: default-src 'self'; script-src 'self' 'unsafe-eval'; style-src 'self' 'unsafe-inline'; frame-src *; img-src * data: blob:; font-src 'self' data:; media-src *; connect-src *
Location: http://localhost/apps/files/
X-Content-Type-Options: nosniff
X-XSS-Protection: 1; mode=block
X-Robots-Tag: none
X-Frame-Options: SAMEORIGIN
X-Download-Options: noopen
X-Permitted-Cross-Domain-Policies: none
Content-Length: 0
Keep-Alive: timeout=5, max=100
Connection: Keep-Alive
Content-Type: text/html; charset=UTF-8
```

**Comment by michaelstingl** [comment]
Yeah, on another instance with the uglier URL's it seems to work.

`GET https://owncloud.mst.works/apps/phoenix/` ==> `HTTP/1.1 200 OK`

Phoenix URL also has no `index.php`. This probably collides with some `.htaccess` magic…

```
…
#### DO NOT CHANGE ANYTHING ABOVE THIS LINE ####

ErrorDocument 403 /core/templates/403.php
ErrorDocument 404 /core/templates/404.php
<IfModule mod_rewrite.c>
  Options -MultiViews
  RewriteRule ^core/js/oc.js$ index.php [PT,E=PATH_INFO:$1]
  RewriteRule ^core/preview.png$ index.php [PT,E=PATH_INFO:$1]
  RewriteCond %{REQUEST_FILENAME} !\.(css|js|svg|gif|png|html|ttf|woff|ico|jpg|jpeg)$
  RewriteCond %{REQUEST_FILENAME} !core/img/favicon.ico$
  RewriteCond %{REQUEST_FILENAME} !/remote.php
  RewriteCond %{REQUEST_FILENAME} !/public.php
  RewriteCond %{REQUEST_FILENAME} !/cron.php
  RewriteCond %{REQUEST_FILENAME} !/core/ajax/update.php
  RewriteCond %{REQUEST_FILENAME} !/status.php
  RewriteCond %{REQUEST_FILENAME} !/ocs/v1.php
  RewriteCond %{REQUEST_FILENAME} !/ocs/v2.php
  RewriteCond %{REQUEST_FILENAME} !/updater/
  RewriteCond %{REQUEST_FILENAME} !/ocs-provider/
  RewriteCond %{REQUEST_URI} !^/.well-known/(acme-challenge|pki-validation)/.*
  RewriteRule . index.php [PT,E=PATH_INFO:$1]
  RewriteBase /
  <IfModule mod_env.c>
    SetEnv front_controller_active true
    <IfModule mod_dir.c>
      DirectorySlash off
    </IfModule>
  </IfModule>
</IfModule>
```

**Comment by michaelstingl** [comment]
Tested with `10.0.10rc1` on the instance with the ugly URLs:

owncloud_phoenix

[LOADING…] stays forever

**Comment by DeepDiver1975** [comment]
> Does it also rely on the index.php in the URL? (default Docker setup has the nicer URL's without index.php)

yes - this is the reason - I need to see how to get this to work ...

**Comment by DeepDiver1975** [comment]
> RewriteCond %{REQUEST_FILENAME} !\.(css|js|svg|gif|png|html|ttf|woff|ico|jpg|jpeg)$

we need to add json file extension here ....

**Comment by michaelstingl** [comment]
Cool, applying https://github.com/owncloud/core/pull/32718 to my `10.0.10 RC1` and changing https://github.com/owncloud/phoenix/pull/225/commits/12950c9c64ab670102a6a5ba448ae3bb30196678 and https://github.com/owncloud/phoenix/pull/225/commits/fa9f14e8cf8e6220414bb3d502d027a890c6e81f in the `phoenix` app made it work in the default Docker setup with clean URL.

But I had to apply the changes manually, I wasn't able to build a new `*.tar` with `make dist`:

```
$ make dist
make: *** No rule to make target `dist'.  Stop.
```

**Comment by michaelstingl** [comment]
Recommendation from @DeepDiver1975 was to use `make -f Makefile.release dist` instead.

This failed later at `cp --parent -R appinfo img README.md CHA…`. `cp --parent` works with GNU-based `cp`, but not with the BSD-based `cp` that macOS uses.

```
cp --parent -R appinfo img README.md CHANGELOG.md LICENSE index.html core/css/uikit.owncloud.css core/js/core.bundle.js node_modules/requirejs/require.js sw.js apps/files/js/files.bundle.js core/gfx/cloud-logo-invert.svg /Users/mstingl_oc/Dev/github.com/owncloud/phoenix/build/dist/phoenix
cp: illegal option -- -
usage: cp [-R [-H | -L | -P]] [-fi | -n] [-apvXc] source_file target_file
       cp [-R [-H | -L | -P]] [-fi | -n] [-apvXc] source_file ... target_directory
make: *** [distdir] Error 64
```

**Comment by michaelstingl** [comment]
@DeepDiver1975 could you rebase? (https://github.com/owncloud/phoenix/pull/230 isn't in here?)

**Comment by DeepDiver1975** [comment]
[phoenix.tar.gz](https://github.com/owncloud/phoenix/files/2414694/phoenix.tar.gz)

**Comment by michaelstingl** [comment]
Is there an easy way to display dummy UI in a way that testers aren't confused?

owncloud_phoenix

**Comment by michaelstingl** [comment]
Could we avoid external hosts? (privacy)

- https://fonts.googleapis.com/icon?family=Material+Icons
- https://fonts.gstatic.com/s/materialicons/v41/flUhRq6tzZclQEJ-Vdg-IuiaDsNcIhQ8tQ.woff2

**Comment by michaelstingl** [comment]
The latest [*.tar](https://github.com/owncloud/phoenix/files/2414694/phoenix.tar.gz) warns me about being offline.

item-0_and_item-0_and_airportextra_and_displaysextra_and_applevolumeextra

**Comment by DeepDiver1975** [comment]
> Is there an easy way to display dummy UI in a way that testers aren't confused?

I can simply remove the whole section

**Comment by michaelstingl** [comment]
> I can simply remove the whole section

👍

**Comment by michaelstingl** [comment]
Hovering top-right icon and username shows strange tooltip:

owncloud_phoenix

**Comment by DeepDiver1975** [comment]
> Could we avoid external hosts? (privacy)
> 
>     * [](https://fonts.googleapis.com/icon?family=Material+Icons)

there is an open issue #228

**Comment by michaelstingl** [comment]
> there is an open issue #228

easy to fix or ignore for early access release?

**Comment by DeepDiver1975** [comment]
> Hovering top-right icon and username shows strange tooltip:

have you set an email address for the user - this is what I see up there.

**Comment by michaelstingl** [comment]
> have you set an email address for the user

yes, fixed it 👍

**Comment by DeepDiver1975** [comment]
[phoenix.tar.gz](https://github.com/owncloud/phoenix/files/2418662/phoenix.tar.gz)

**Comment by michaelstingl** [comment]
Just found out, that's already working when the app is placed in an alternative apps folder. 😍 😍 😍 

Seems the app icon is hardcoded to the `apps` directory? Other assets too?

files_-_owncloud_enterprise_edition

(No showstopper for the pre-alpha early release from my pov)

**Comment by DeepDiver1975** [comment]
> Just found out, that's already working when the app is placed in an alternative apps folder. heart_eyes heart_eyes heart_eyes

interesting ... but looks like a core issue - not a phoenix issue. need to have a look one day ....

**Pull request merged**